In [14]:
from pykrx import stock
from pykrx import bond

today="20251201"

tickers_kospi_default = stock.get_market_ticker_list(today, market="KOSPI")
tickers_kospi = stock.get_market_ticker_list(today, market="KOSPI")
tickers_kosdaq = stock.get_market_ticker_list(today, market="KOSDAQ")
tickers_konex = stock.get_market_ticker_list(today, market="KONEX")
tickers_all = stock.get_market_ticker_list(today, market="ALL")

#tickers_del_count = len(tickers_del)

print(f"--- KRX 상장 종목 개수 ({today} 기준) ---")
print(f"1. KOSPI (기본값):  {len(tickers_kospi_default)} 개") 
print(f"2. KOSPI (명시):    {len(tickers_kospi)} 개")
print(f"3. KOSDAQ:          {len(tickers_kosdaq)} 개")
print(f"4. KONEX:           {len(tickers_konex)} 개")
print(f"5. 전체 시장 (ALL): {len(tickers_all)} 개")

# sanity check (KOSPI + KOSDAQ + KONEX == ALL)
# print(f"합산 검증: {len(tickers_kospi) + len(tickers_kosdaq) + len(tickers_konex)} 개")

--- KRX 상장 종목 개수 (20251201 기준) ---
1. KOSPI (기본값):  958 개
2. KOSPI (명시):    958 개
3. KOSDAQ:          1812 개
4. KONEX:           117 개
5. 전체 시장 (ALL): 2887 개


In [20]:
ticker="036570"

for ticker in stock.get_market_ticker_list():
        company = stock.get_market_ticker_name(ticker)
print(company)

흥아해운


In [21]:
ticker = "036570" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)

엔씨소프트


In [ ]:
for ticker in stock.get_market_ticker_list():
    company = stock.get_market_ticker_name(ticker)
    print(f"티커: {ticker}, 회사명: {company}")

In [18]:
ticker = "005930"
start_date = "20250801"
end_date = "20251201" # end_date necessary

company_name = stock.get_market_ticker_name(ticker)
print(company_name)
df = stock.get_market_ohlcv(start_date, end_date, ticker)
print(df.head(3))
print(df.tail(3))

삼성전자
               시가     고가     저가     종가       거래량       등락률
날짜                                                        
2025-08-01  70200  72000  68800  68900  24038534 -3.501401
2025-08-04  69500  70200  68700  69700  14780072  1.161103
2025-08-05  71000  71500  69700  69900  14392903  0.286944
                시가      고가      저가      종가       거래량       등락률
날짜                                                            
2025-11-27  104100  105500  102700  103500  16453004  0.680934
2025-11-28  103800  103800  100500  100500  15292277 -2.898551
2025-12-01  102000  102800   99900  100800  10905526  0.298507


get_market_ohlcv() 함수에 frequency 파라미터(d/m/y)를 추가할 수 있습니다. d는 일자, m은 월, y는 연도를 의미합니다. 다음은 20200810부터 20201212까지의 삼성전자 월별 OHLCV 데이터를 출력합니다.


In [19]:
df = stock.get_market_ohlcv(start_date, end_date, ticker, 'm') 
# 월봉을 보려면, 'm'을 사용해야 한다.
# 만약 waring을 제거하려면

print(df.head)

<bound method NDFrame.head of                 시가      고가     저가      종가        거래량
날짜                                                  
2025-08-31   70200   72400  68300   69700  282943885
2025-09-30   68400   86200  67500   83900  409841865
2025-10-31   84900  108600  84700  107500  484630984
2025-11-30  106900  112400  94500  100500  472946077
2025-12-31  102000  102800  99900  100800   10905526>


In [20]:
# delete warning
import warnings
from pykrx import stock
warnings.filterwarnings('ignore', category=FutureWarning)

df = stock.get_market_ohlcv(start_date, end_date, ticker, 'm')
print(df.head)

warnings.filterwarnings('default', category=FutureWarning)

<bound method NDFrame.head of                 시가      고가     저가      종가        거래량
날짜                                                  
2025-08-31   70200   72400  68300   69700  282943885
2025-09-30   68400   86200  67500   83900  409841865
2025-10-31   84900  108600  84700  107500  484630984
2025-11-30  106900  112400  94500  100500  472946077
2025-12-31  102000  102800  99900  100800   10905526>


In [47]:
df = stock.get_market_ohlcv(start_date, end_date, ticker, adjusted=False)
print(df)

                시가      고가      저가      종가       거래량           거래대금   등락률
날짜                                                                       
2025-08-01   70200   72000   68800   68900  24038534  1680885451444 -3.50
2025-08-04   69500   70200   68700   69700  14780072  1030549721600  1.16
2025-08-05   71000   71500   69700   69900  14392903  1012313553300  0.29
2025-08-06   69200   69400   68300   68800  12542415   863115544500 -1.57
2025-08-07   70800   71000   69700   70500  15008213  1055402519950  2.47
...            ...     ...     ...     ...       ...            ...   ...
2025-11-25  101400  101400   97800   99300  16110054  1602730841700  2.69
2025-11-26  100500  102900   99300  102800  21314975  2159034953350  3.52
2025-11-27  104100  105500  102700  103500  16453004  1713246599050  0.68
2025-11-28  103800  103800  100500  100500  15292277  1548884266354 -2.90
2025-12-01  102000  102800   99900  100800  10905526  1101995992050  0.30

[81 rows x 7 columns]


get_stock_ticker_list() 함수와 함께 사용하면 전종목의 OHLCV를 가져올 수 있습니다. 한 번에 너무 많은 정보를 요청하면 KRX 서버가 차단할 수 있으므로 time 모듈을 사용해서 1초 지연 후 요청하도록 코드를 작성했습니다.
```python
from pykrx import stock
import time

TICKERS_LIST = stock.get_market_ticker_list("20251202", market='ALL')
print(f"총 {len(TICKERS_LIST)}개의 종목 조회를 시작합니다.")

for i, ticker in enumerate(TICKERS_LIST):
    try:
        start_date = "20251201"
        end_date = "20251202"
        company = stock.get_market_ticker_name(ticker)
        df = stock.get_market_ohlcv(start_date, end_date, ticker) 
        
        print(f"\n[{i+1}/{len(TICKERS_LIST)}] {company} ({ticker}) 데이터:")
        print(df) # 2일치 데이터이므로 head() 대신 전체 출력

    except Exception as e:
        print(f"오류 발생: {company} ({ticker}) - {e}")

    time.sleep(1)

print("\n--- 모든 종목 조회 완료 ---")
```


In [18]:
#get_market_ohlcv(): kospi 전종목 시세를 df로 반환, index ticker
#단일 종목, index date
ticker = "005930"
company_name = stock.get_market_ticker_name(ticker)
print(company_name)
df = stock.get_market_ohlcv("20251201", "20251202", ticker)
print(df)

#전종목, index ticker
df = stock.get_market_ohlcv("20251201")
print(df.head(3))

삼성전자
                시가      고가      저가      종가       거래량       등락률
날짜                                                            
2025-12-01  102000  102800   99900  100800  10905526  0.298507
2025-12-02  101200  103500  101000  103400  13619200  2.579365
          시가    고가    저가    종가     거래량        거래대금   등락률          시가총액
티커                                                                    
095570  4700  4845  4670  4775  237140  1131943487  3.02  216081924225
006840  9080  9080  8990  9000     782     7036860  0.11  119228049000
027410  4040  4065  3945  3970   92601   368975274 -1.61  379995660270


In [19]:
#get_market_price_change: 입력된 기간 동안의 가격 변동 정보, 가격 변동폭
ticker = "005930"
company_name = stock.get_market_ticker_name(ticker)
print(company_name)

df = stock.get_market_price_change("20251101", "20251106")
print(df.head())

# krx web server, 전일 종가를 시작가로 반환.
df = stock.get_market_ohlcv("20251101", "20251106", "005930")
print(df)

삼성전자
            종목명      시가      종가   변동폭   등락률      거래량         거래대금
티커                                                               
095570   AJ네트웍스    4300    4320    20  0.47   989021   4187829019
006840    AK홀딩스    9800    9480  -320 -3.27    61028    579727850
027410      BGF    3805    3820    15  0.39   601973   2246115448
282330   BGF리테일  103600  105600  2000  1.93   173377  17943891800
138930  BNK금융지주   13810   14950  1140  8.25  4916492  70886136890
                시가      고가      저가      종가       거래량       등락률
날짜                                                            
2025-11-03  106900  111500  106500  111100  30110066  3.348837
2025-11-04  111800  112400  104900  104900  30450281 -5.580558
2025-11-05  101000  102000   96700  100600  44843020 -4.099142
2025-11-06  103700  103800   98800   99200  28655689 -1.391650


In [20]:
from pykrx import stock
from pykrx import bond

df = stock.get_market_ohlcv("20251101", "20251106", "000020")
print(df.head())
print(df.columns)
df = stock.get_market_price_change("20251101", "20251106")
print(df.head())

              시가    고가    저가    종가     거래량       등락률
날짜                                                  
2025-11-03  6280  6320  6140  6150  105026 -1.757188
2025-11-04  6180  6270  6150  6260   85341  1.788618
2025-11-05  6280  6280  6110  6250   96904 -0.159744
2025-11-06  6250  6310  6170  6240   65941 -0.160000
Index(['시가', '고가', '저가', '종가', '거래량', '등락률'], dtype='object')
            종목명      시가      종가   변동폭   등락률      거래량         거래대금
티커                                                               
095570   AJ네트웍스    4300    4320    20  0.47   989021   4187829019
006840    AK홀딩스    9800    9480  -320 -3.27    61028    579727850
027410      BGF    3805    3820    15  0.39   601973   2246115448
282330   BGF리테일  103600  105600  2000  1.93   173377  17943891800
138930  BNK금융지주   13810   14950  1140  8.25  4916492  70886136890


In [11]:
print(df.loc["282330"]) # ticker index일 때만 성립

종목명          BGF리테일
시가           103600
종가           105600
변동폭            2000
등락률            1.93
거래량          173377
거래대금    17943891800
Name: 282330, dtype: object


In [23]:
#ticker = "005930" # ticker index일 때만 성립"
#company_name = stock.get_market_ticker_name(ticker)
#print(company_name)

#get_market_ohlcv(): kospi 전종목 시세를 df로 반환, index ticker
#df = stock.get_market_ohlcv("20251101", "20251106", "000020")

#get_market_price_change: 입력된 기간 동안의 가격 변동 정보, 가격 변동폭(오늘 종가 - 전일 종가)
# 등락률 변동폭/전일종가*100, 거래량 매도된 주식수 + 매수된 주식 수 = 체결된 총 주식수
# 거래대금: (거래량*평균거래가격)/일일 거래대금((시가+종가)/2), 평균거래가격: 거래대금/거래
#df = stock.get_market_price_change("20251101", "20251106")
df = stock.get_market_price_change("20251101", "20251106", market="KOSDAQ")
print(df.head(5))

          종목명     시가     종가  변동폭   등락률      거래량         거래대금
티커                                                          
060310     3S   1770   1700  -70 -3.95  1106980   1912618943
054620    APS   4060   4085   25  0.62   138251    542411980
265520  AP시스템  19900  19140 -760 -3.82   805180  15647435365
211270   AP위성  12460  11560 -900 -7.22   279886   3375532610
139050   BF랩스   2805   2805    0  0.00        0            0


```
          종목명     시가     종가  변동폭   등락률      거래량         거래대금
티커                                                          
060310     3S   1770   1700  -70 -3.95  1106980   1912618943
054620    APS   4060   4085   25  0.62   138251    542411980
265520  AP시스템  19900  19140 -760 -3.82   805180  15647435365
211270   AP위성  12460  11560 -900 -7.22   279886   3375532610
139050   BF랩스   2805   2805    0  0.00        0            0
```
**등락률 변동폭/전일종가*100, 거래량 매도된 주식수 + 매수된 주식 수 = 체결된 총 주식수\
거래대금: (거래량*평균거래가격)/일일 거래대금((시가+종가)/2), 평균거래가격: 거래대금/거래**\
시가,전일종가	종가	변동폭(종가-전일종가)	등략률(변동폭/전일종가)	거래량	거래대금	평균거래가격1(거래대금/거래량)	거래대금1(거래량*평균거래가격1)	평균거래가격2(시가+종가)/2	거래대금(거래량*평균거래가격2)\
12460	11560	-900.00	-7.22%	279886	3375532610	12060.38391	3375532610	12010	3361430860

#### 2.1.1.5 종목별 DIV/DPS/BPS/PER/EPS 조회

| 지표 | 전체이름 | 수식 |핵심 | 의미|지표 해석 (일반적) |
|:---:|:---:|:---:|:---:|:---:| ---|
|EPS(수익성)|주당 순이익 (Earnings Per Share)|$$\text{EPS} = \frac{\text{당기순이익}}{\text{총 발행 주식 수}}$$|주식 1주당 순이익 창출 능력|높을수록 기업의 수익성이 좋음|
|PER(수익성)|주가 수익 비율 (Price Earnings Ratio)|$$\text{PER} = \frac{\text{현재 주가}}{\text{EPS}}$$|주가가 순이익 대비 몇 배인지 (가장 대중적인 수익성 평가)|낮을수록 저평가로 간주될 수 있음|
|BPS(자산가치)|주당 순자산 (Book-value Per Share)|$$\text{BPS} = \frac{\text{총 순자산 (자본)}}{\text{총 발행 주식 수}}$$|주식 1주당 순자산 가치 (회사의 장부 가치)|높을수록 기업의 재무 건전성이 좋음|
|PBR(자산가치)|주가 순자산 비율 (Price Book-value Ratio)|$$\text{PBR} = \frac{\text{현재 주가}}{\text{BPS}}$$|주가가 순자산 대비 몇 배인지 (대표적인 자산 평가)|1 미만일 경우, 장부 가치보다 주가가 낮음 (저평가 가능성)|
|DPS(배당)|주당 배당률 (Dividend Per Shared)|$$\text{DPS} = \frac{\text{총 배당금 지급액}}{\text{총 발행 주식 수}}$$|주식 1주당 주주에게 지급되는 현금 금액|꾸준히 증가, 유지, 재무상태 안정적, 주주환원의지|
|DIV(배당)|배당수익률 (Dividend Yield)|$$\text{DIV} = \frac{\text{DPS}}{\text{현재 주가}} \times 100$$|투자 원금 대비 현금 배당으로 얻는 수익률|높을수록 배당 매력이 높음|

**순자산(총자산 - 총부채)**\
PER & PBR: 둘 다 주가를 평가하는 지표이지만, $\text{PER}$은 수익 창출 능력을, $\text{PBR}$은 보유 자산 가치를 기준으로 평가합니다. 가치 투자자는 보통 $\text{PER}$과 $\text{PBR}$이 낮은 종목을 선호합니다.\
DPS & DIV: $\text{DPS}$는 배당금의 절대적인 금액을, $\text{DIV}$는 그 배당금이 주가 대비 어느 정도의 수익률을 제공하는지를 나타냅니다.\
EPS & BPS: 이 두 지표는 $\text{PER}$과 $\text{PBR}$의 분모가 되는 기업의 기초 체력 지표입니다. 이 수치가 높을수록 기업의 기초가 튼튼하다고 볼 수 있습니다.

In [45]:
df = stock.get_market_fundamental("20210108")
print(df.head(2))
df = stock.get_market_fundamental("20210104", market="KOSDAQ")
print(df.head(2))
#print(df.loc[ticker]) # df의 ticker가 index일 때만 성립"


          BPS    PER   PBR   EPS   DIV  DPS
티커                                         
095570   6802   4.62  0.67   982  6.61  300
006840  62448  11.69  0.41  2168  2.96  750
          BPS  PER   PBR  EPS  DIV  DPS
티커                                     
060310    745  0.0  3.03    0  0.0    0
054620  10530  0.0  0.76    0  0.0    0
삼성전자
              BPS    PER   PBR   EPS   DIV   DPS
날짜                                              
2021-01-04  37528  26.22  2.21  3166  1.71  1416
2021-01-05  37528  26.50  2.24  3166  1.69  1416


In [54]:
ticker = "005930" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)
df = stock.get_market_fundamental("20210104", "20210108", "005930")
print(df.head(2))
df = stock.get_market_fundamental("20251203", "20251203", "005930")
print(df.head(2))
#df = stock.get_market_fundamental("20251203", "005930") #error
df = stock.get_market_fundamental("20200101", "20200430", "005930", freq="m") #Warning
print(df.head(2))
#BPS Book-value per share, 주당 순자산
#PER Price Earning Ratio, 주가 수익 비율
#PBR Price Book-value Ratio, 주당 순자산 비율
#EPS Earning per share, 주당 순이익
#DIV Dividend yield, 배당 수익률
#DPS Dividend per share, 주당 배당률

삼성전자
              BPS    PER   PBR   EPS   DIV   DPS
날짜                                              
2021-01-04  37528  26.22  2.21  3166  1.71  1416
2021-01-05  37528  26.50  2.24  3166  1.69  1416
              BPS    PER  PBR   EPS   DIV   DPS
날짜                                             
2025-12-03  57951  21.11  1.8  4950  1.38  1446
              BPS   PER   PBR   EPS   DIV   DPS
날짜                                             
2020-01-31  35342  8.54  1.56  6461  2.57  1416
2020-02-29  35342  8.85  1.62  6461  2.48  1416


C:\Users\ppjm7\AppData\Local\Programs\Python\Python312\Lib\site-packages\pykrx\stock\stock_api.py:46: FutureWarning: 'M' is deprecated and will be removed in a future version, please use 'ME' instead.
  df = df.resample('M').apply(how)


In [55]:
# delete warning
import warnings
from pykrx import stock
warnings.filterwarnings('ignore', category=FutureWarning)
ticker = "005930" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)
df = stock.get_market_fundamental("20200101", "20200430", ticker, freq="m") #Warning
print(df.head(2))

warnings.filterwarnings('default', category=FutureWarning)

삼성전자
              BPS   PER   PBR   EPS   DIV   DPS
날짜                                             
2020-01-31  35342  8.54  1.56  6461  2.57  1416
2020-02-29  35342  8.85  1.62  6461  2.48  1416


#### 2.1.1.7 일자별 거래실적 추이 (거래대금)
지정된 기간 동안의 일자별로 정렬된 (순매수) 거래량을 얻어옵니다. 당일자 최종 매매내역은 오후 6시 이후에 제공됩니다. 다음은 삼성전자 (005930)의 2021-01-15부터 2021-01-22까지의 거래대금을 얻어옵니다.

In [79]:
#순매수액, 기호(-) 매수우위=순매도
df = stock.get_market_trading_value_by_date("20210115", "20210122", "005930")
print(df.head())

                    기관합계         기타법인            개인         외국인합계  전체
날짜                                                                   
2021-01-15 -440769209300  25442287800  661609085600 -246282164100   0
2021-01-18   42323535000  22682344800   14829121700  -79835001500   0
2021-01-19   95523053500  -3250422500 -173484213300   81211582300   0
2021-01-20 -364476214000  22980632900  430115581000  -88619999900   0
2021-01-21  -60637506300 -27880854000  250285510000 -161767149700   0


`매수` 혹은 `매도` 금액을 조회하고 싶다면 `on` 파라미터에 입력할 수 있습니다. 입력하지 않는다면 `순매수` 금액이 조회됩니다.

In [58]:
#총매도액
df = stock.get_market_trading_value_by_date("20210115", "20210122", "005930", on='매도')
print(df.head())


                     기관합계         기타법인             개인         외국인합계  \
날짜                                                                    
2021-01-15  1505250243300  19525756900   925426195700  525029741764   
2021-01-18  1353223733500  22132983500  1883422627200  456996648400   
2021-01-19  1205081446400  28155090000  1800734616800  407371601300   
2021-01-20  1147190346800  10015618200   643413644500  405119540200   
2021-01-21   878411497500  41184362100   851899126900  439714802000   

                       전체  
날짜                         
2021-01-15  2975231937664  
2021-01-18  3715775992600  
2021-01-19  3441342754500  
2021-01-20  2205739149700  
2021-01-21  2211209788500  


In [59]:
#총매수액, 총매수액=총매도액(1조5천) + 순매수액(- 4천4백억) = 1조 6백억 
df = stock.get_market_trading_value_by_date("20210115", "20210122", "005930", on='매수')
print(df.head())


                     기관합계         기타법인             개인         외국인합계  \
날짜                                                                    
2021-01-15  1064481034000  44968044700  1587035281300  278747577664   
2021-01-18  1395547268500  44815328300  1898251748900  377161646900   
2021-01-19  1300604499900  24904667500  1627250403500  488583183600   
2021-01-20   782714132800  32996251100  1073529225500  316499540300   
2021-01-21   817773991200  13303508100  1102184636900  277947652300   

                       전체  
날짜                         
2021-01-15  2975231937664  
2021-01-18  3715775992600  
2021-01-19  3441342754500  
2021-01-20  2205739149700  
2021-01-21  2211209788500  


세 번째 파라미터에 `KOSPI`, `KOSDAQ`, `KONEX`를 사용하면 각 시장의 거래대금을 조회합니다. `ALL`을 입력하면 세 개의 시장을 합친 통합 거래대금을 조회합니다.

In [64]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "KOSPI")
print(df.head())

                     기관합계         기타법인             개인         외국인합계  전체
날짜                                                                     
2021-01-15 -1414745885546  54444293672  2113924037705 -753622445831   0
2021-01-18  -278880716957 -26004926379   514299140387 -209413497051   0
2021-01-19   593956459208  21472281148 -1025418915468  409990175112   0
2021-01-20 -1234485992694  34510184945  1436793223994 -236817416245   0
2021-01-21  -292666343147 -13168420832   -62476631241  368311395220   0


In [67]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "KOSPI", "KOSDAQ")
print(df.head())

                     기관합계         기타법인             개인         외국인합계  전체
날짜                                                                     
2021-01-15 -1530847752261  62945760367  2248352385215 -780450393321   0
2021-01-18  -592898052807 -27623514934   489211555972  131310011769   0
2021-01-19   544152851138  21821266493  -967296086698  401321969067   0
2021-01-20 -1228718682989  32493389130  1442397406214 -246172112355   0
2021-01-21  -286837721397 -18614779712   -59748014721  365200515830   0


In [68]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "KOSPI", "KOSDAQ", "KONEX")
print(df.head())

                     기관합계         기타법인             개인         외국인합계  전체
날짜                                                                     
2021-01-15 -1531939778041  63218516217  2248639407480 -779918145656   0
2021-01-18  -596785557357 -27007739039   491867173077  131926123319   0
2021-01-19   544093068088  21715369418  -967438346413  401629908907   0
2021-01-20 -1227963978269  32218594740  1443334999369 -247589615840   0
2021-01-21  -284395914072 -19087585227   -61502732221  364986231520   0


In [69]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "ALL")
print(df.head())

                     기관합계         기타법인             개인         외국인합계  전체
날짜                                                                     
2021-01-15 -1509110430886  36143566359  2370005512792 -897038648265   0
2021-01-18  -285038978281 -56972180359   525180437507 -183169278867   0
2021-01-19   638265287634  -2223711739 -1300259840511  664218264616   0
2021-01-20 -1019968701538   2325703232   986953036131   30689962175   0
2021-01-21  -476620709128 -74745939889   300165858321  251200790696   0


In [63]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True)
print(df.head())

                     기관합계         기타법인             개인         외국인합계  전체
날짜                                                                     
2021-01-15 -1536570309441  63110174617  2251672617980 -778212483156   0
2021-01-18  -601428111357 -27000808439   494341183227  134087736569   0
2021-01-19   544654406338  21787409868  -968965427363  402523611157   0
2021-01-20 -1227642472619  32139813590  1444113501769 -248610842740   0
2021-01-21  -284899892322 -19072459127   -61503500921  365475852370   0


In [70]:
df = stock.get_market_trading_value_by_date("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True, detail=True)
print(df.head())

                     금융투자            보험            투신           사모  \
날짜                                                                   
2021-01-15  -991569370471 -101879326615  -73654664921 -47097103944   
2021-01-18  -234901037556  -72240710086  101791257920 -22451555756   
2021-01-19  1095649201830  -16407303019 -132856710155  12116061569   
2021-01-20  -642974676550  -63074125224  -93392192931  -2355882539   
2021-01-21   354360314201 -107524405663  -12601355617  21898580074   

                      은행          기타금융           연기금         기타법인  \
날짜                                                                  
2021-01-15  202104737955   -4226413080 -520248168365  63110174617   
2021-01-18   29622542645    9331964150 -412580572674 -27000808439   
2021-01-19    1776752770   -1933723125 -413689873532  21787409868   
2021-01-20  -17889264145   -1573759250 -406382571980  32139813590   
2021-01-21 -113225122905 -158268629075 -269539273337 -19072459127   

                       개인

#### 2.1.1.8 일자별 거래실적 추이 (거래량)
일자별 시장의 거래량을 조회합니다. 당일자 최종 매매내역은 오후 6시 이후에 제공됩니다. 다음은 삼성전자 (005930)의 2021-01-15부터 2021-01-22까지의 거래량을 얻어옵니다.

In [72]:
ticker = "005930" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "005930")
print(df.head())

삼성전자
               기관합계    기타법인       개인    외국인합계  전체
날짜                                               
2021-01-15 -5006115  288832  7485785 -2768502   0
2021-01-18   505669  262604   151228  -919501   0
2021-01-19  1139258  -34023 -2044543   939308   0
2021-01-20 -4157919  262408  4917655 -1022144   0
2021-01-21  -712099 -321732  2890389 -1856558   0


In [73]:
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "005930", on='매도')
print(df.head())

                기관합계    기타법인        개인    외국인합계        전체
날짜                                                       
2021-01-15  16944750  216653  10371182  5899224  33431809
2021-01-18  15709256  258096  21942253  5318346  43227951
2021-01-19  13912581  323382  20956376  4702705  39895044
2021-01-20  13121791  114341   7346474  4628521  25211127
2021-01-21  10093381  473635   9714878  5036117  25318011


In [74]:
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "005930", on='매수')
print(df.head())

                기관합계    기타법인        개인    외국인합계        전체
날짜                                                       
2021-01-15  11938635  505485  17856967  3130722  33431809
2021-01-18  16214925  520700  22093481  4398845  43227951
2021-01-19  15051839  289359  18911833  5642013  39895044
2021-01-20   8963872  376749  12264129  3606377  25211127
2021-01-21   9381282  151903  12605267  3179559  25318011


In [75]:
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "KOSPI")
print(df.head())

                기관합계     기타법인        개인     외국인합계  전체
날짜                                                   
2021-01-15 -20393142  8435634  29119751 -17162243   0
2021-01-18  -5700054 -1198498  15316328  -8417776   0
2021-01-19   7216278  -246496 -24395243  17425461   0
2021-01-20 -23038683  -793906  31606917  -7774328   0
2021-01-21 -18443990 -7082091   8365421  17160660   0


In [76]:
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True)
print(df.head())

                기관합계     기타법인        개인     외국인합계  전체
날짜                                                   
2021-01-15 -26571037  8455599  37942108 -19826670   0
2021-01-18 -65039501  -757841  39922005  25875337   0
2021-01-19 -41855511  4320588  31709703   5825220   0
2021-01-20 -23038880 -2562184  38031657 -12430593   0
2021-01-21 -38539026 -8798430  38195538   9141918   0


In [77]:
df = stock.get_market_trading_volume_by_date("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True, detail=True)
print(df.head())

                금융투자       보험        투신       사모        은행     기타금융      연기금  \
날짜                                                                             
2021-01-15 -30909240 -1400206   2245052 -2307623  11071023   102571 -5372614   
2021-01-18 -70318619 -1883526   9224883    10962   3203493   394649 -5671343   
2021-01-19 -24047200  -161246 -11083467 -1024629    216630  -150452 -5605147   
2021-01-20 -10187734 -1024404  -3957101  -523064   -556527   163402 -6953452   
2021-01-21 -13155430 -3602153  -2107802  -742226  -3466644 -8481152 -6983619   

               기타법인        개인       외국인   기타외국인  전체  
날짜                                                   
2021-01-15  8455599  37942108 -19793169  -33501   0  
2021-01-18  -757841  39922005  25583326  292011   0  
2021-01-19  4320588  31709703   5392346  432874   0  
2021-01-20 -2562184  38031657 -12710114  279521   0  
2021-01-21 -8798430  38195538   8855139  286779   0  


#### 2.1.1.9 투자자별 거래실적 추이 (거래대금)
지정된 기간 동안의 투자자별로 정렬된 (순매수) 거래대금을 얻어옵니다. 당일자 최종 매매내역은 오후 6시 이후에 제공됩니다. 다음은 삼성전자 (005930)의 2021-01-15부터 2021-01-22까지의 거래대금을 얻어옵니다.

In [78]:
df = stock.get_market_trading_value_by_investor("20210115", "20210122", "005930")
print(df.head())

                  매도             매수           순매수
투자자구분                                            
금융투자   2580964135000  2309054317700 -271909817300
보험      153322228800    44505136200 -108817092600
투신      258073006600   150715203700 -107357802900
사모       65167773900    60862926800   -4304847100
은행        3369626100     4004806100     635180000


#### 2.1.1.7 일자별 거래실적 추이 (거래대금)
지정된 기간 동안의 일자별로 정렬된 (순매수) 거래량을 얻어옵니다. 당일자 최종 매매내역은 오후 6시 이후에 제공됩니다. 다음은 삼성전자 (005930)의 2021-01-15부터 2021-01-22까지의 거래대금을 얻어옵니다.

In [81]:
#순매수액, 기호(-) 매수우위=순매도
df = stock.get_market_trading_value_by_date("20210115", "20210122", "005930")
print(df.head())

                    기관합계         기타법인            개인         외국인합계  전체
날짜                                                                   
2021-01-15 -440769209300  25442287800  661609085600 -246282164100   0
2021-01-18   42323535000  22682344800   14829121700  -79835001500   0
2021-01-19   95523053500  -3250422500 -173484213300   81211582300   0
2021-01-20 -364476214000  22980632900  430115581000  -88619999900   0
2021-01-21  -60637506300 -27880854000  250285510000 -161767149700   0


In [82]:
df = stock.get_market_trading_value_by_investor("20210115", "20210122", "KOSPI")
print(df.head())

                  매도             매수           순매수
투자자구분                                            
금융투자   9827334289654  9294592831462 -532741458192
보험      912820396542   560818697065 -352001699477
투신     1790231574897  1421181450288 -369050124609
사모      830445404788   665802837480 -164642567308
은행       58624439870    37109603010  -21514836860


In [83]:
df = stock.get_market_trading_value_by_investor("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True)
print(df.head())

                   매도              매수           순매수
투자자구분                                              
금융투자   15985568261831  15006116511544 -979451750287
보험      1219035502445    757575677208 -461459825237
투신      2235561259511   1799363743367 -436197516144
사모       999084910863    846067212945 -153017697918
은행       886226324790    936210985810   49984661020


#### 2.1.1.10 투자자별 거래실적 추이 (거래량)
지정된 기간 동안의 투자자별로 정렬된 (순매수) 거래량을 얻어옵니다. 당일자 최종 매매내역은 오후 6시 이후에 제공됩니다. 다음은 삼성전자 (005930)의 2021-01-15부터 2021-01-22까지의 거래량을 얻어옵니다.

In [84]:
df = stock.get_market_trading_volume_by_investor("20210115", "20210122", "005930")
print(df.head())

             매도        매수      순매수
투자자구분                             
금융투자   29455909  26450600 -3005309
보험      1757287    509535 -1247752
투신      2950680   1721970 -1228710
사모       745727    696135   -49592
은행        38675     46394     7719


In [85]:
df = stock.get_market_trading_volume_by_investor("20210115", "20210122", "KOSPI")
print(df.head())

              매도         매수       순매수
투자자구분                                
금융투자   137969209  127697577 -10271632
보험      15737709    8577242  -7160467
투신      46872846   34307243 -12565603
사모      20780475   16342937  -4437538
은행       2236667     632814  -1603853


In [86]:
df = stock.get_market_trading_volume_by_investor("20210115", "20210122", "KOSPI", etf=True, etn=True, elw=True)
print(df.head())

               매도          매수        순매수
투자자구분                                   
금융투자   1857447354  1660620713 -196826641
보험       29594468    19872165   -9722303
투신       69348190    60601427   -8746763
사모       31673292    26585281   -5088011
은행       44279242    51690814    7411572


#### 2.1.1.11 투자자별 순매수 상위종목
`get_market_net_purchases_of_equities` 함수는 순매수 금액 순서로 정렬된 종목 리스트를 반환합니다. 다음은 코스피 시장에서 2021-01-15 부터 2021-01-22 까지 개인의 순매수 금액이 높은 순서대로 종목을 정렬해서 반환합니다.\
투자자 항목에는 다음 중 하나를 설정할 수 있습니다.\
금융투자 / 보험 / 투신 / 사모 / 은행 / 기타금융 / 연기금 / 기관합계 / 기타법인 / 개인 / 외국인 / 기타외국인 / 전체

In [10]:
from pykrx import stock
from pykrx import bond

df = stock.get_market_net_purchases_of_equities("20251115", "20251201", "KOSPI", "외국인")
print(df.head())

            종목명     매도거래량     매수거래량   순매수거래량         매도거래대금         매수거래대금  \
티커                                                                           
329180  HD현대중공업   2410575   2680652   270077  1349347094000  1503424321000   
015760     한국전력  14307873  17149256  2841383   713533240600   855922634225   
064350     현대로템   2354382   2921541   567159   426474278800   523621196050   
326030   SK바이오팜   1082055   1674688   592633   144385393650   222167327850   
004370       농심    159419    298139   138720    73210797250   137589334750   

             순매수거래대금  
티커                    
329180  154077227000  
015760  142389393625  
064350   97146917250  
326030   77781934200  
004370   64378537500  


#### 2.1.1.12 종목별 시가 총액 조회
`get_market_cap` 함수는 특정 일자의 종목별 시가총액/거래량/거래대금/상장주식수/외국인보유주식수를 조회합니다.\
시가총액과 거래대금의 단위는 원입니다. 외국인보유주식수는 D-2 (영업일 기준)일 까지 유효한 값이 출력되며, D-1일은 0으로 표기됩니다.

In [11]:
df = stock.get_market_cap("20200625")
print(df.head())

            종가             시가총액       거래량          거래대금       상장주식수
티커                                                                 
005930   51900  309831714345000  18541624  970494909540  5969782550
000660   84300   61370599369500   3397112  287640892200   728002365
207940  815000   53924475000000    163339  133594217000    66165000
035420  269500   44268984952500   1196267  325381436500   164263395
068270  316000   42640845660000    918369  291688043500   134939385


In [15]:
ticker = "068270" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)

셀트리온


#### 2.1.1.13 일자별 시가 총액 조회
`get_market_cap` 함수는 특정 종목의 일자별 시가총액을 조회합니다. 거래량/거래대금/상장주식수가 함께 조회됩니다.
```python
df = stock.get_market_cap("20190101", "20190131", "005930")
print(df.head())
```

In [18]:
df = stock.get_market_cap("20251101", "20251231", "005930")
print(df.tail())

                       시가총액       거래량           거래대금       상장주식수
날짜                                                              
2025-11-28  594923611161000  15292277  1548884266354  5919637922
2025-12-01  596699502537600  10905526  1101995992050  5919637922
2025-12-02  612090561134800  13649487  1401469615424  5919637922
2025-12-03  618602162849000  14697927  1541179456000  5919637922
2025-12-04  622153945602200  11931145  1243000736272  5919637922


In [28]:
tickers = stock.get_etf_ticker_list("20251101")
print(tickers[:10])


['495710', '466810', '457930', '487750', '445690', '0112X0', '465780', '442260', '0001P0', '159800']
Empty DataFrame
Columns: []
Index: []


In [27]:
df = stock.get_etf_ohlcv_by_date("20251104", "20251231", "495710")
print(df)

                 NAV     시가     고가     저가     종가  거래량      거래대금    기초지수
날짜                                                                     
2025-11-04  51401.37  51415  51415  51415  51415    1     51415  107.05
2025-11-05  51401.75  51415  51415  51415  51415    1     51415  107.05
2025-11-06  51397.30  51415  51415  51415  51415    1     51415  107.04
2025-11-07  51403.03  51415  51415  51415  51415    1     51415  107.05
2025-11-10  51404.91  51420  51420  51420  51420    1     51420  107.05
2025-11-11  51407.47  51420  51420  51420  51420    5    257100  107.06
2025-11-12  51405.79  51425  51425  51425  51425    1     51425  107.06
2025-11-13  51401.73  51430  51430  51430  51430    1     51430  107.06
2025-11-14  51409.20  51425  51425  51425  51425    1     51425  107.08
2025-11-17  51412.64  51425  51425  51425  51425    1     51425  107.09
2025-11-18  51415.87  51430  51430  51430  51430    1     51430  107.10
2025-11-19  51419.68  51405  51430  51405  51430    2    102835 

In [31]:
df = stock.get_etf_portfolio_deposit_file("159800")
print(df.head())

           계약수         금액     비중
티커                              
005930  4705.0  491672500  27.77
000660   564.0  311328000  17.17
105560   326.0   42934200   2.36
005380   136.0   36244000   2.17
034020   432.0   33868800   1.92


In [33]:
ticker = "034020" 
company_name = stock.get_market_ticker_name(ticker)
print(company_name)

두산에너빌리티


In [ ]:
tickers = stock.get_etf_ticker_list("20251201")
for ticker in tickers:
    print(stock.get_etf_ticker_name(ticker))

In [ ]:
tickers = stock.get_etf_ticker_list("20251201")
print(tickers)

In [ ]:
tickers = stock.get_etf_ticker_list("20251201")

for ticker in tickers:
    name = stock.get_etf_ticker_name(ticker)
    print(f"{ticker} : {name}")

In [46]:
from pykrx import stock
import pandas as pd

tickers = stock.get_etf_ticker_list("20251201")

data = []

#for ticker in tickers[:5]:    # 앞에서 5개만
for ticker in tickers:    # 앞에서 5개만
    name = stock.get_etf_ticker_name(ticker)
    data.append([ticker, name])
    #print(f"{ticker} : {name}")

df = pd.DataFrame(data, columns=["ticker", "name"])
print(df)

      ticker                     name
0     495710  BNK 26-06 특수채(AAA이상)액티브
1     466810              BNK 2차전지양극재
2     457930            BNK 미래전략기술액티브
3     487750              BNK 온디바이스AI
4     445690              BNK 주주가치액티브
...      ...                      ...
1042  0000Y0   HK 26-12 회사채(AA-이상)액티브
1043  215620             HK S&P코리아로우볼
1044  0113P0               HK 머니마켓액티브
1045  391670             HK 베스트일레븐액티브
1046  472920        HK 종합채권(AA-이상)액티브

[1047 rows x 2 columns]


In [52]:
df = stock.get_etf_ohlcv_by_date("20251201", "20251231", "495710")
print(df.head())

                 NAV     시가     고가     저가     종가  거래량      거래대금    기초지수
날짜                                                                     
2025-12-01  51446.60  51465  51465  51465  51465    1     51465  107.16
2025-12-02  51450.87  51465  51465  51465  51465    8    411720  107.17
2025-12-03  51453.04  51440  51465  51440  51465  201  10339465  107.17
2025-12-04  51470.22  51480  51480  51480  51480    1     51480    0.00


In [53]:
etf_ticker = "495710" 
etf_name = stock.get_etf_ticker_name(etf_ticker)
print(etf_name)

BNK 26-06 특수채(AAA이상)액티브


In [55]:
df = stock.get_stock_major_changes("005930")
df.tail()

상호변경전 상호변경후 업종변경전 업종변경후  액면변경전 액면변경후                   대표이사변경전  \
날짜                                                                          
2025-03-19     -     -     -     -      0     0                       한종희   
2025-03-25     -     -     -     -      0     0                    한종희전영현   
2025-03-25     -     -     -     -      0     0  Jong-heeHanYoung-HyunJun   
2025-11-28     -     -     -     -      0     0             Young-hyunJun   
2025-11-28     -     -     -     -      0     0                       전영현   

                             대표이사변경후  
날짜                                    
2025-03-19                    한종희전영현  
2025-03-25                       전영현  
2025-03-25             Young-HyunJun  
2025-11-28  Young-hyunJunTae-moonRoh  
2025-11-28                    전영현노태문